workflow/executor/executor.go

```diff
@@ -780,25 +791,27 @@ func (we *WorkflowExecutor) CaptureScriptResult(ctx context.Context) error {
 	return nil
 }
 
+func (we *WorkflowExecutor) reportOutputsCompleted(ctx context.Context) error {
+	return we.reportResult(ctx, wfv1.NodeResult{}, map[string]string{common.LabelKeyReportOutputsCompleted: "true"})
+}
```

Comment:
is this adding an empty NodeResult? It looks like you're doing this in the defer in wait.go, meaning you add a NodeResult with Outputs at the end of the function and then add an empty NodeResult?

Reply:
My bad, I should be able to pass null instead of an empty noderesult, I think. It should upsert.

Reply:
So, what I was thinking was this: remove all of the existing changes in wait.go and all of the report methods here. In the createTaskResult() function set this label to false. Then in wait.go do a `defer wfExecutor.FinalizeOutput()`. That method just upserts the label set to true. See any reason that can't work or isn't good design? It seems simple anyway.

Reply:
or let me know if you had something else in mind and I can try to advise based on that...I would for one like to remove the fact that two different reportOutputs-type methods are called at the end of the wait function

Reply:
@juliev0 Do you mean combining the `wfExecutor.ReportOutputs` and `wfExecutor.ReportOutputsCompleted` functions? If so, I will need to add some state to avoid having to pass `logArtifacts` directly to the deferred `ReportOutputs` function. I'm thinking it would make most sense to add the state to  `executor.WorkflowExecutor` in the same way that `error` exists there.

Reply:
Is the purpose of `reportOutputsCompleted()` to simply add the label? If that's the case then why does it need to be added to `reportOutputs`?

Reply:
Yes, it is to add the label. As you requested previously, it's mostly there as a backup incase there is a random failure that isn't caught. I'm happy to remove the label setting from `reportOutputs` since it should be set either way in the defer.

As per why we don't just go back to calling reportOutputs in `SaveLogs`, I didn't like it there since it seemed like the responsibility of `SaveLogs` should not include reporting outputs (since not all outputs are logs). This confused me when I was trying to debug the issue since I didn't expect it there.

Reply:
(if what i'm saying doesn't make sense I can try to clarify better - but it's basically the idea I was trying to capture [here](https://github.com/argoproj/argo-workflows/pull/11947/files#r1373975383))

Reply:
"As per why we don't just go back to calling reportOutputs in SaveLogs, I didn't like it there since it seemed like the responsibility of SaveLogs should not include reporting outputs (since not all outputs are logs). This confused me when I was trying to debug the issue since I didn't expect it there." - Oh, yes. Great point. i didn't realize that. Then let's keep doing that.

Reply:
"I'm happy to remove the label setting from reportOutputs since it should be set either way in the defer." Thanks so much. I think in that case `reportResult()` can remain as it was before (just reports NodeResults). (I apologize if I'm being too controlling. I'm still learning how to strike the right balance as a reviewer. Feel free to let me know if I make a bad suggestion.) :)

Reply:
No problem. I'm implementing the change now.

Reply:
looks good! thank you!